autogen/oai/completion.py

```diff
@@ -185,28 +208,31 @@ def _get_response(cls, config: Dict, raise_on_ratelimit_or_timeout=False, use_ca
         """
         config = config.copy()
         openai.api_key_path = config.pop("api_key_path", openai.api_key_path)
+
+        model_name = config.get("model")
+        mapped_model_name = cls.llm_lite_mapping.get(model_name, model_name)  # Fallback to original if not found
+        config["model"] = mapped_model_name
+
         key = get_key(config)
         if use_cache:
-            response = cls._cache.get(key, None)
-            if response is not None and (response != -1 or not raise_on_ratelimit_or_timeout):
-                # print("using cached response")
-                cls._book_keeping(config, response)
-                return response
-        openai_completion = (
-            openai.ChatCompletion
-            if config["model"] in cls.chat_models or issubclass(cls, ChatCompletion)
-            else openai.Completion
-        )
+            try:
+                response = cls._cache.get(key, None)
+                if response is not None and (response != -1 or not raise_on_ratelimit_or_timeout):
+                    # print("using cached response")
+                    cls._book_keeping(config, response)
+                    return response
+            except:
+                pass
         start_time = time.time()
         request_timeout = cls.request_timeout
         max_retry_period = config.pop("max_retry_period", cls.max_retry_period)
         retry_wait_time = config.pop("retry_wait_time", cls.retry_wait_time)
         while True:
             try:
                 if "request_timeout" in config:
-                    response = openai_completion.create(**config)
+                    response = litellm.completion(**config)
```

Comment:
This change will raise many questions. @qingyun-wu @victordibia have commented before. @victordibia 's question in #45 is one example.

Reply:
Reading through the ticket. I like the idea of adding documentation, guiding people on how to use other models.

Reply:
@krrishdholakia yes, please!

I'm planning on making another YouTube video about using OS models in AutoGen. Hopefully that'll also make it easier for people to do.

Reply:
@mberman84 I like where you are going with this. I'm currently watching your AutoGen tutorial video but need Claude 2. I'm not a programmer and need all the help I can get. I appreciate you and everyone else contributing to this PR!